File: androidApp/src/main/java/com/cacd2/cacdgame/android/features/search/SearchLexicalEmptyView.kt
package com.cacd2.cacdgame.android.features.search

import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Surface
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cacd2.cacdgame.android.AppTheme
import com.cacd2.cacdgame.android.R
import com.cacd2.cacdgame.android.fonts

@Composable
fun SearchLexicalEmptyView(modifier: Modifier = Modifier) {
    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Image(painter = painterResource(id = R.drawable.lexicon_empty),
            contentDescription = "")
        Text(
            text = stringResource(id = R.string.lexical_empty_list),
            modifier = Modifier.padding(20.dp),
            textAlign = TextAlign.Center,
            style =
            TextStyle(
                fontFamily = fonts,
                fontSize = 18.sp,
                fontWeight = FontWeight.SemiBold
            )
        )
    }
}

@Preview
@Composable
fun SearchLexicalEmptyPreview() {
    AppTheme {
        Surface {
            SearchLexicalEmptyView(modifier = Modifier.fillMaxSize())
        }
    }
}

File: iosApp/iosApp/Features/Search/SearchLexicalEmptyView.swift
//
//  SearchLexicalEmptyView.swift
//  CACDGAME
//
//  Created by François Dabonot on 06/12/2023.
//  Copyright © 2023 CACD2. All rights reserved.
//

import SwiftUI

struct SearchLexicalEmptyView: View {
    var body: some View {
        VStack(alignment: .center, content: {
            R.image.lexicon_empty.image
            Text(R.string.localizable.lexical_empty_list.localizedKey)
                .multilineTextAlignment(.center)
                .padding()
                .font(R.font.poppinsSemiBold.font(size: 18))
        })
    }
}

#Preview {
    SearchLexicalEmptyView()
}
